test/crontab/functional_test.exs

```diff
@@ -220,4 +220,54 @@ defmodule Crontab.FunctionalTest do
       assert Crontab.DateChecker.matches_date?(cron_expression, @start_date) == @matches_now
     end
   end
+
+  tests_find_date_tz_changes = [
+    # DST Shift with extra hour
+    {"30 2 * * *", [:earlier], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+02:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [:later], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+01:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [:earlier, :later], "Europe/Zurich", ~N[2024-10-27 01:00:01],
+     ["2024-10-27T02:30:00+02:00", "2024-10-27T02:30:00+01:00", "2024-10-28T02:30:00+01:00"]},
+    {"30 2 * * *", [], "Europe/Zurich", ~N[2024-10-27 01:00:01], ["2024-10-28T02:30:00+01:00"]},
```

Comment:
I reckon it might be faster to ask you about this: the screenshot shows the unit to advance is `:hour`, instead of the expected `:day`. 'cos if it's `:hour`, `shift/4` is correct (ignore ambiguity opts, 'cos hourly run expectation). But this cron expression is "at 2:30 daily", so advancement should be `:day`.

Any idea?
 
![image](https://github.com/user-attachments/assets/10056760-9a55-494a-a64b-a531d689df88)

Reply:
@shaolang I think we fist have to be clear about the `Crontab.DateChecker`. The scheduler should find all dates efficiently for instances where the date checker evaluates to truthy.